docs/java-rest/high-level/apis/shrink_index.asciidoc

```diff
@@ -0,0 +1,90 @@
+[[java-rest-high-shrink-index]]
+=== Shrink Index API
+
+[[java-rest-high-shrink-index-request]]
+==== Resize Request
+
+The Shrink API requires a `ResizeRequest` instance.
+A `ResizeRequest` requires two string arguments:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[shrink-index-request]
+--------------------------------------------------
+<1> The target index (first argument) to shrink the source index (second argument) into
+
+==== Optional arguments
+The following arguments can optionally be provided:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[shrink-index-request-timeout]
+--------------------------------------------------
+<1> Timeout to wait for the all the nodes to acknowledge the index is opened
+as a `TimeValue`
+<2> Timeout to wait for the all the nodes to acknowledge the index is opened
+as a `String`
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[shrink-index-request-masterTimeout]
+--------------------------------------------------
+<1> Timeout to connect to the master node as a `TimeValue`
+<2> Timeout to connect to the master node as a `String`
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[shrink-index-request-waitForActiveShards]
+--------------------------------------------------
+<1> The number of active shard copies to wait for before the shrink index API
+returns a response, as an `int`.
+<2> The number of active shard copies to wait for before  the shrink index API
```

Comment:
nit: extra space

Reply:
how do you detect this? do you have extra powers? :)

Reply:
My wife would say that I am very meticulous about certain things (and perfectly loose on others)